### PR TITLE
Rename redis to valkey

### LINKED ISSUE #8249: Rename redis by valkey everwhere
### Description and goal

We've switched from redis to valkey, but we keep calling it redis. We should stop doing that

### Acceptance criteria

```
$ git grep -i redis
$
```

- [ ] Redis is replaced by valkey everywhere

### PR DESCRIPTION
**Pull request by EthanPERRUZZA**
Fixes #8249.

### CODE REVIEW

**Comment by emersion** [comment]
Nit: we're not using the "chore" prefixes from conventional commits. Instead we use a prefix to indicate the component, e.g. "editoast".

Other nit: "rename XXX by YYY" doesn't sound right to me, maybe "rename XXX to YYY" would be better?

**Comment by EthanPERRUZZA** [comment]
> Nit: we're not using the "chore" prefixes from conventional commits. Instead we use a prefix to indicate the component, e.g. "editoast".

Should I put editoast: front: docker: or editoast, front, docker: or maybe all: or multiple: ?

**Comment by emersion** [comment]
"editoast, front, docker:" is common. "all:" is something I'd do (and e.g. Go does) but I haven't seen it in OSRD yet.

**Comment by EthanPERRUZZA** [comment]
> "editoast, front, docker:" is common. "all:" is something I'd do (and e.g. Go does) but I haven't seen it in OSRD yet.

I've look through the commit and the first one is the one used ;)

**Review by Khoyo — COMMENTED** [review]
Nice! I'm not approving because we'll need to update the charts at the same time.

Nit: it really is rename to: https://english.stackexchange.com/questions/100296/which-preposition-to-use-with-rename

**Comment by emersion** [comment]
Gentle ping about this:

> Other nit: "rename XXX by YYY" doesn't sound right to me, maybe "rename XXX to YYY" would be better?

File: editoast/src/map/mod.rs [hunk 1]
```diff
@@ -11,47 +11,47 @@ pub use self::layer_cache::get_layer_cache_prefix;
...
 }
 
 /// Invalidates all map layers of a specific infra
 ///
 /// # Arguments
 ///
-/// * `redis_pool` - Pool to use to connect to the redis
-/// * `layers` - Layers to invalidate
+/// * `valkey` - Pool to use to connect to the valkey
+/// * `layers` - Layers to valkey
```
Comment (woshilapin) [inline editoast/src/map/mod.rs]: Shouldn't this stay `invalidate`? Why the change?

```suggestion
/// * `layers` - Layers to invalidate
```

File: editoast/src/views/timetable.rs [hunk 2]
```diff
@@ -287,7 +287,7 @@ async fn conflicts(
     }
 
     let db_pool = app_state.db_pool_v2.clone();
-    let redis_client = app_state.redis.clone();
+    let valkey_client: std::sync::Arc<crate::ValkeyClient> = app_state.valkey.clone();
```
Comment (woshilapin) [inline editoast/src/views/timetable.rs]: Why do you need the type to be explicit now? Can't you keep it like it was before?

**Review by woshilapin — APPROVED** [review]
Thanks for the hard work.

**Comment by flomonster** [comment]
@Khoyo Why do you want to rename all occurrences of redis to valkey? We always use redis in production and from editoast's point of view, the service used is redis. The lib name is redis.

> [!NOTE]
> I'm okay with:
> - renaming redis to valkey in the osrd-compose since the image used is valkey
> - using valkey in our CI

**Comment by emersion** [comment]
Would be nice to switch to valkey in production as well.

**Comment by Khoyo** [comment]
> We always use redis in production and from editoast's point of view, the service used is redis. The lib name is redis.

As far as the open source side of the project is concerned, we use an opensource software named valkey and formerly known as redis. I believe from the point of view of editoast, the service used should be the open source one, not a closed source fork.

The valkey crate name is currently unofficially reserved by the redis-rs maintainers, and I believe we *should* switch to it if it becomes necessary.

> using valkey in our CI

I believe that was @emersion first PR :) https://github.com/OpenRailAssociation/osrd/pull/7515

---

Now, as far as production is concerned, we use neither redis nor valkey but Amazon ElastiCache.

Seeing the current move toward supporting valkey in favor of redis at aws (see the elasticache [supported engine versions](https://docs.aws.amazon.com/AmazonElastiCache/latest/dg/SelectEngine.html), and [this blogpost](https://aws.amazon.com/blogs/database/amazon-elasticache-and-amazon-memorydb-announce-support-for-valkey/)), we should probably move there too.

**Comment by EthanPERRUZZA** [comment]
> Gentle ping about this:
> 
> > Other nit: "rename XXX by YYY" doesn't sound right to me, maybe "rename XXX to YYY" would be better?

Sorry, I had completly forgotten, thanks for the ping ;)

**Review by flomonster — DISMISSED** [review]
LGTM.

We should have the osrd-chart (+ deployments) PRs ready.

**Review by SharglutDev — APPROVED** [review]

**Review by Khoyo — APPROVED** [review]
